src/routes/(main-app)/+layout.svelte

```diff
@@ -3,21 +3,21 @@
   import { enhance } from "$app/forms";
   import {page} from "$app/stores";
   import "$lib/styles/app.css";
-  import type { SubmitFunction } from "./$types";
-  
+  import type { SubmitFunction } from "@sveltejs/kit";
+  import type { LayoutProps } from "./$types";
+  import { invalidate, invalidateAll } from "$app/navigation";
   //import { enhance } from '$app/forms';
-  import type { ActionData } from './$types.js';
-  //export let data;
-  export let form: ActionData;
-  export let closed = false;
+
+  let { data }: LayoutProps = $props();
```

Comment:
You also need to change this line.
```suggestion
  let { data, children }: LayoutProps = $props();
```